src/inheritAttributes.js

```diff
@@ -50,16 +50,24 @@ export const buildBaseUrls = (referenceUrls, baseUrlElements) => {
  */
 export const getSegmentInformation = (adaptationSet) => {
   const segmentTemplate = findChildren(adaptationSet, 'SegmentTemplate')[0];
-  const segmentTimeline =
-    segmentTemplate && findChildren(segmentTemplate, 'SegmentTimeline')[0];
   const segmentList = findChildren(adaptationSet, 'SegmentList')[0];
+  const segmentUrls = segmentList && findChildren(segmentList, 'SegmentURL');
   const segmentBase = findChildren(adaptationSet, 'SegmentBase')[0];
+  const segmentTimelineNode = segmentList ? segmentList : segmentTemplate;
+  const segmentTimeline =
+    segmentTimelineNode && findChildren(segmentTimelineNode, 'SegmentTimeline')[0];
 
   return {
     template: segmentTemplate && getAttributes(segmentTemplate),
     timeline: segmentTimeline &&
                      findChildren(segmentTimeline, 'S').map(s => getAttributes(s)),
-    list: segmentList && getAttributes(segmentList),
+    list: segmentList &&
+            shallowMerge(getAttributes(segmentList),
+              {
+                segmentUrls: segmentUrls &&
+                  segmentUrls.map(s =>
```

Comment:
you could move this `map` to `findChildren(segmentList, 'SegmentURL').map(...`, then this just becomes `{ segmentUrls }`